src/Microsoft.AspNetCore.Routing/DefaultEndpointDataSourceBuilder.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Builder;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    internal class DefaultEndpointDataSourceBuilder : EndpointDataSourceBuilder
+    {
+        public IApplicationBuilder ApplicationBuilder { get; set; }
```

Comment:
I think we should follow a similar approach to what we did in MVC to get the app builder... OR.... use https://github.com/aspnet/Hosting/blob/master/src/Microsoft.AspNetCore.Hosting/Builder/ApplicationBuilderFactory.cs#L11 - that's what it's there for.

Reply:
OK, I'm fine with leaving this as-is right now. We'll sort this out when we've got more clarity.